### PR TITLE
Transform improvements

### PR DESCRIPTION
**Pull request by maZahaca**
- Moved transforms by different files
- Added missing lanGoose definitions for transforms

:goose:

### CODE REVIEW

**Comment by maZahaca** [comment]
@jifeon review it please

File: lib/Parser.js [hunk 1]
```diff
@@ -290,15 +289,22 @@ Parser.prototype = {
...
     },
 
     /**
-     * @see {@link Transformations#addTransformation}
+     * @deprecated use addTransform instead
      */
-    addTransformation: function() {
-        return this._transforms.addTransformation.apply(this._transforms, arguments)
+    addTransformation: function(type, transform) {
+        return this.addTransform(type, transform);
```
Comment (jifeon) [inline lib/Parser.js]: I would add `debug` message here as well, kinda warning

File: lib/Transforms.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+const debug = require('debug')('Transforms');
+const Storage = require('./Storage');
+const transformsFactory = require('./transforms/transformsFactory');
+
+function Transforms(options) {
```
Comment (jifeon) [inline lib/Transforms.js]: you are using es6 classes bellow, why to not use them here?

File: lib/transforms/TransformCombine.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+const Transform = require('./Transform');
+const _ = require('lodash');
+
+class TransformCombine extends Transform {
+    doTransform() {
+        const fields = this._options.fields || [];
+        const type = this._options.dataType || 'string';
+        const resultValue = [];
+        _.each(fields, field => {
```
Comment (jifeon) [inline lib/transforms/TransformCombine.js]: pls use `Array#forEach`

File: lib/transforms/TransformJoin.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+ * Perform click to specified selector on the page
+ */
+
+'use strict';
+
+const Transform = require('./Transform');
+
+class TransformJoin extends Transform {
+    doTransform() {
+        const glue = this._options.glue || ' ';
```
Comment (jifeon) [inline lib/transforms/TransformJoin.js]: so `glue` cannot be empty string? `'' || ' ' === ' '`

File: lib/transforms/TransformMatch.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+'use strict';
+
+const Transform = require('./Transform');
+
+class TransformMatch extends Transform {
+    doTransform() {
+        const re = this._options.re;
+        const index = this._options.index || 0;
+
+        matches = (this._value || '').match(RegExp.apply(null, re));
```
Comment (jifeon) [inline lib/transforms/TransformMatch.js]: and if `this._value` is zero? I think need to use `String` here

File: lib/transforms/TransformReplace.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+const Transform = require('./Transform');
+
+class TransformReplace extends Transform {
+    doTransform() {
+        const re = this._options.re;
+        if (!Array.isArray(re)) {
+            throw new Error('You must pass an array as `re` to `replace` transform');
+        }
+
+        return (this._value || '').replace(RegExp.apply(null, re), this._options.to);
```
Comment (jifeon) [inline lib/transforms/TransformReplace.js]: the same here

File: lib/transforms/TransformSplit.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+'use strict';
+
+const Transform = require('./Transform');
+
+class TransformSplit extends Transform {
+    doTransform() {
+        const index = this._options.index || 0;
+        const dataType = this._options.dataType || 'string';
+        let separator = this._options.separator;
+        separator = Array.isArray(separator) ? RegExp.apply(null, separator) : separator || ',';
```
Comment (jifeon) [inline lib/transforms/TransformSplit.js]: separator cannot be empty string?

File: lib/transforms/TransformMatch.js (hunk 5 above)
Comment (maZahaca) [inline lib/transforms/TransformMatch.js]: Which options are available to do that?

**Comment by maZahaca** [comment]
@jifeon Can you have a look on [TransformPick](https://github.com/redco/goose-parser/blob/1154c7c066104bb12cc37ffa46d263d31865d407/lib/transforms/TransformPick.js) and [TransformGet](https://github.com/redco/goose-parser/blob/1154c7c066104bb12cc37ffa46d263d31865d407/lib/transforms/TransformGet.js). Seems they are duplicates.

**Comment by jifeon** [comment]
@maZahaca is it ready for re-review?

**Comment by maZahaca** [comment]
I fixed issues, you mentioned before.
For now I just added one more transform.

I've tried to run it on real data, and some transform fail. Gonna fix that issue

**Pull request merged**